src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -689,6 +689,207 @@ protected override void ProcessRecord()
         }
 
         #endregion Overrides
+
+        /// <summary>
+        /// Adds UserName, Description, BinPath and Delay Autostart info to the ServiceController object.
+        /// </summary>
+        /// <param name="service"></param>
+        /// <returns></returns>
+        private PSObject AddProperties(ServiceController service) {
+            NakedWin32Handle hScManager = IntPtr.Zero;
+            NakedWin32Handle hService = IntPtr.Zero;
+            int lastError = 0;
+            PSObject serviceAsPSObj = PSObject.AsPSObject(service);
+            try
+            {
+                hScManager = NativeMethods.OpenSCManagerW(
+                    lpMachineName: service.MachineName,
+                    lpDatabaseName: null,
+                    dwDesiredAccess: NativeMethods.SC_MANAGER_CONNECT
+                );
+                if (IntPtr.Zero == hScManager) {
+                    lastError = Marshal.GetLastWin32Error();
+                    Win32Exception exception = new Win32Exception(lastError);
+                    WriteNonTerminatingError(
+                        service,
+                        service.MachineName,
+                        exception,
+                        "ComputerAccessDenied",
+                        ServiceResources.ComputerAccessDenied,
+                        ErrorCategory.PermissionDenied);
+                }
+                hService = NativeMethods.OpenServiceW(
+                    hScManager,
+                    service.ServiceName,
+                    NativeMethods.SERVICE_QUERY_CONFIG
+                );
+                if (IntPtr.Zero == hService) {
+                    lastError = Marshal.GetLastWin32Error();
+                    Win32Exception exception = new Win32Exception(lastError);
+                    WriteNonTerminatingError(
```

Comment:
I think rather than writing a non-terminating error for each property that fails to be retrieved, it might be better to just write one non-terminating error if any fail.